packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,313 @@
+  color: ${uiColors.gray.light2};
+`;
+
+const grayBorder = css`
+  height: ${borderHeight}px;
+  background-color: ${uiColors.gray.light2};
+  position: relative;
+  // Makes border overlap with box-model for TabTitle components
+  margin-top: -${borderHeight}px;
+  pointer-events: none;
+`;
+
+const greenBorder = css`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  background-color: ${uiColors.green.base};
+  transition: 150ms transform ease-in-out, 150ms width ease-in-out 10ms;
+`;
+
+const focusedStyle = css`
+  background-color: ${uiColors.blue.base};
+`;
+
+function isTab(
+  element: React.ReactNode,
+): element is React.ReactElement<TabProps, typeof Tab> {
+  if (!(element != null && typeof element === 'object' && 'type' in element)) {
+    return false;
+  }
+
+  return (element.type as any).displayName === 'Tab';
+}
+
+interface TabsProps {
+  /**
+   * Content that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when a tab is selected.
+   */
+  onChange?: React.ReactEventHandler;
+
+  /**
+   * Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: string | null;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+
+  /**
+   * HTML Element that wraps title in Tab List.
+   */
+  as?: React.ElementType<any>;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs onChange={() => execute callback onChange}>
+  <Tab value='tab-1' title='First Tab'>Tab 1</Tab>
+  <Tab value='tab-2' title='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.onChange Callback to be executed when a Tab is selected.
+ * @param props.selected Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className Classname applied to Tabs container.
+ * @param props.as HTML Element that wraps title in Tab List.
+ */
+function Tabs({
+  children,
+  onChange,
+  selected,
+  className,
+  as = 'button',
+  ...rest
+}: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([]);
+  const tabListRef = useRef<HTMLDivElement>(null);
+
+  const childrenArray = React.Children.toArray(children) as Array<
+    React.ReactElement
+  >;
+
+  useEffect(() => {
+    const currIdx = childrenArray.reduce(
+      (acc, child, index) => {
+        if (!child) {
+          return acc;
+        }
+
+        if (selected || activeTab) {
+          return [activeTab, selected].includes(child.props.value)
+            ? [...acc, index]
+            : acc;
+        }
+
+        if (child.props.default) {
+          return [...acc, index];
+        }
+
+        return acc;
+      },
+      [] as Array<number>,
+    );
```

Comment:
This looks like it's building up an array of indexes, but you're only ever using one. Would this work the same way?

```ts
const currrentIndex = childrenArray.findIndex(child => child && (
  [activeTab, selected].includes(child.props.value) || child.props.default
));
```

Reply:
^ this breaks it

Reply:
Oh my logic was off. What about...

```ts
const currentIndex = childrenArray.findIndex(child => {
  if (!child) {
    return false;
  }

  if (activeTab || selected) {
    return [activeTab, selected].includes(child.props.value);
  }
  return child.props.default;
});
```

Reply:
🎆

Reply:
Got it :)